### DISCUSSION TITLE
Ongoing BG Notification no longer shows notification age

### DISCUSSION THREAD

**Discussion by richard-dennehy** [NightscoutFoundation/xDrip#3578] [Bug reports]
Hi,

I recently updated my xDrip version and noticed the timestamp has disappeared from the BG notification. It was useful to be able to glance at the notification and know how old the BG reading was.

![image](https://github.com/user-attachments/assets/aecf3583-fdfd-4ffd-a47e-a0db33ea4aa4)

This is because the target SDK has recently been increased to 24 (Android N), and `Notification.Builder.setWhen` no longer displays the provided timestamp by default as of Android N (see [here](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/app/Notification.java#4289)).

I've tested on an emulator and adding `b.setShowWhen(true)` after [this line](https://github.com/NightscoutFoundation/xDrip/blob/master/app/src/main/java/com/eveningoutpost/dexdrip/utilitymodels/Notifications.java#L653) fixes it. I could raise a PR for this if you want.

![image](https://github.com/user-attachments/assets/073e9b6c-58c9-4307-be96-c3b9e4ed34dc)

Thanks

**Comment by Navid200** [comment]
Yes, please open a PR and it will be reviewed.
Thanks a lot for reporting this as well as all the work you have put in already,
